### PR TITLE
Add reduce option to F.cross_covariance

### PR DESCRIPTION
**Pull request by delta2323**
This PR adds `reduce` option to `F.CrossCovariance` and `F.cross_covariance`. Related to #2558

### CODE REVIEW

File: chainer/functions/loss/cross_covariance.py [hunk 1]
```diff
@@ -36,37 +42,64 @@ def forward(self, inputs):
...
+    If it is ``'half_frobenius_norm'``, it holds the half of the
+    Frobenius norm (i.e. L2 norm of a matrix flattened to a vector)
+    of the covarianct matrix.
+
     Args:
         y (Variable): Variable holding a matrix where the first dimension
             corresponds to the batches.
         z (Variable): Variable holding a matrix where the first dimension
             corresponds to the batches.
+        recude (str): Reduction option. Its value must be either
```
Comment (unnonouno) [inline chainer/functions/loss/cross_covariance.py]: reduce (typo)

File: tests/chainer_tests/functions_tests/loss_tests/test_cross_covariance.py [hunk 2]
```diff
@@ -12,38 +12,60 @@
...
     def check_forward(self, y_data, z_data):
         y = chainer.Variable(y_data)
         z = chainer.Variable(z_data)
-        loss = functions.cross_covariance(y, z)
-        self.assertEqual(loss.data.shape, ())
+        loss = functions.cross_covariance(y, z, self.reduce)
+
+        row = y_data.shape[1]
+        col = z_data.shape[1]
+        if self.reduce == 'half_frobenius_norm':
```
Comment (unnonouno) [inline tests/chainer_tests/functions_tests/loss_tests/test_cross_covariance.py]: `self.assertEqual(loss.data.shape, self.gloss.shape)` is enough

File: chainer/functions/loss/cross_covariance.py (hunk 1 above)
Comment (delta2323) [inline chainer/functions/loss/cross_covariance.py]: Fixed

File: tests/chainer_tests/functions_tests/loss_tests/test_cross_covariance.py (hunk 2 above)
Comment (delta2323) [inline tests/chainer_tests/functions_tests/loss_tests/test_cross_covariance.py]: Fixed

File: chainer/functions/loss/cross_covariance.py [hunk 3]
```diff
@@ -10,11 +10,17 @@ class CrossCovariance(function.Function):
 
     """Cross-covariance loss."""
 
-    def __init__(self):
+    def __init__(self, reduce='half_frobenius_norm'):
```
Comment (unnonouno) [inline chainer/functions/loss/cross_covariance.py]: Is it `half_squared_sum`?

**Comment by unnonouno** [comment]
LGTM
